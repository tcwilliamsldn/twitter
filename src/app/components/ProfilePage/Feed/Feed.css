.feed {
    margin: 12px;
    padding: 0;
    background-color: #ffffff;
}
