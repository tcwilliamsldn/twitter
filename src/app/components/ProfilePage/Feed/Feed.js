import React from 'react';
import Tweet from './Tweet/Tweet';
import styles from './Feed.css';

const Feed = ({ tweets }) => (
    <div className={styles.feed}>
        {tweets.map(props => <Tweet {...props} key={props.id} />)}
    </div>
);

export default Feed;
