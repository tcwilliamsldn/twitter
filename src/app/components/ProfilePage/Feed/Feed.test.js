import Feed from './Feed';
import React from 'react';
import { shallow } from 'enzyme';

describe('Feed', function() {
    it('renders a users Tweets', function() {
        const tweet = {
            text: 'Hello World',
            created_at: 'Fri Jul 20 02:17:02 +0000 2018',
            id: '123',
            user: 'tom'
        };

        const component = shallow(
            <Feed tweets={[tweet, { ...tweet, id: '456' }]} />
        );

        expect(component).toMatchSnapshot();
    });
});
