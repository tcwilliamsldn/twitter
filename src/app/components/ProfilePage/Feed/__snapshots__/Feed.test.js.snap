// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Feed renders a users Tweets 1`] = `
<div
  className="feed"
>
  <Tweet
    created_at="Fri Jul 20 02:17:02 +0000 2018"
    id="123"
    key="123"
    text="Hello World"
    user="tom"
  />
  <Tweet
    created_at="Fri Jul 20 02:17:02 +0000 2018"
    id="456"
    key="456"
    text="Hello World"
    user="tom"
  />
</div>
`;
